lua/neogit/buffers/log_view/init.lua

```diff
@@ -122,7 +122,7 @@ function M:open()
           end
         end,
         ["d"] = function()
-          if not config.ensure_integration("diffview") then
```

Comment:
instead of a boolean param, can we do an opts table, like `{ required = x }`? Not a huge fan of naked boolean params

Reply:
Yes. Not entirely sure about the existing name for the function either.

`ensure_`, at least the way I've encountered and used the name refers to a function which does a check and in some way *ensures* a value, such as generating one if it doesn't exist, ensures a column is *present* in a database etc.

This is essentially a query or check function, especially when adding a `{ required = true }`

Perhaps `query_integration(name) -> bool` and the have the caller do an error or fallback, with better context, such as `diffview integration needs to be enabled for "diff head"`

Reply:
I like the idea of leaving it up to the caller. That removes the need for the `{ required = true }` param altogether, and... wait for it... we could call it... `check_integration(name)`. Because it's a check.. for the integration. :P

Reply:
Great. Often great solutions come from stepping back and looking at the larger picture, rather than adding yet another bool argument

Reply:
Made the change to check and provided some more descriptive error messages that show a notification.

Just fixed a `call("string", true, false, true true)` situation where a big async function begun with an if statement, that either allocated a scratch buffer and did a thing, or did a thing directly. Just pried it apart into two functions, as the bools were never dynamic but always hardcoded anyway

Reply:
Yeahhh. Thats something I remember from.. I think Uncle Bob... early on in my career. If you are controlling a functions behaviour externally with boolean flags... congrats, you have two functions :P